prow/pod-utils/decorate/podspec_test.go

```diff
@@ -1893,6 +1893,360 @@ func TestProwJobToPod(t *testing.T) {
 				},
 			},
 		},
+		{
+			podName: "pod",
```

Comment:
This test case is for undecorated jobs with multiple containers which is something I think we should explicitly forbid during config validation. Our recommendation is that all new ProwJobs use pod utility decoration and Prow defaults to warning about undecorated jobs in `checkconfig`, we only allow jobs to be undecorated for backwards compatibility. 
If we later find that we need/want to allow undecorated jobs with multiple containers we can easily remove the restriction, but I think it will promote best practices and reduce complexity.